src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/Threads.java

```diff
@@ -241,7 +241,12 @@ public JavaThread owningThreadFromMonitor(ObjectMonitor monitor) {
                         return thread;
                      }
                 }
-                throw new InternalError("We should have found a thread that owns the anonymous lock");
+                // We should have found the owner. However, the code can run concurrently with
+                // Java code and locking state can change at any time. This code is not
```

Comment:
The second sentence is not accurate as things are not running concurrently and cannot change state at any time. Suggestion:
```
// We should have found the owner, however, as the VM could be in any state, including the middle
// of performing GC, it is not always possible to do so. Just return null if we can't locate it.
```